src/mpid/ch4/shm/ipc/xpmem/xpmem_post.h

```diff
@@ -47,7 +55,9 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_IPC_xpmem_close_mem(MPIDI_IPC_xpmem_mem_seg_t
 
     MPIR_Assert(mem_seg.seg_ptr);
 
```

Comment:
Assert should be inside the guard otherwise it will always fail when XPMEM is disabled, right?